addons/mail/static/src/utils/test_utils.js

```diff
@@ -568,19 +585,34 @@ async function start(param0 = {}) {
         null,
         () => testEnv.messagingBus.trigger('will_show_home_menu')
     );
-    testEnv.modelManager.start(testEnv);
-    /**
-     * Create the messaging singleton record.
-     */
-    testEnv.messaging = testEnv.models['mail.messaging'].create();
-    testEnv.messaging.start().then(() =>
-        testEnv.messagingInitializedDeferred.resolve()
-    );
 
     const result = { env: testEnv, mockServer, widget };
 
-    if (waitUntilMessagingInitialized) {
-        // env key only accessible after MessagingService has started
+    const createMessaging = () => {
+        Object.assign(testEnv, {
+            modelManager: new ModelManager(testEnv),
+        });
+
+        testEnv.modelManager.start(testEnv);
+        /**
+         * Create the messaging singleton record.
+         */
+        testEnv.messaging = testEnv.models['mail.messaging'].create();
+        testEnv.messaging.start().then(() =>
+            testEnv.messagingInitializedDeferred.resolve()
+        );
+        testEnv.messagingCreatedPromise.resolve();
+    };
+
+    if (messagingBeforeCreationDeferred) {
+        messagingBeforeCreationDeferred.then(() => createMessaging());
+    } else {
+        createMessaging();
+    }
```

Comment:
Would read better if the default value of `messagingBeforeCreationDeferred` was a resolved promise, and we just `then` on it no matter what.